source/lib/cdk-textract-stack.ts

```diff
@@ -973,6 +1191,15 @@ export class CdkTextractStack extends cdk.Stack {
       }
     );
 
+    // add kendra index id to lambda environment in case of DUS+Kendra mode
+    if(props.enableKendra && createKendraIndexCustomResource!=null){
```

Comment:
Why do we have this case again? Can we not nest this under the initial if props.enableKendra condition?

Reply:
Since we need the apiProcessor, and other lambdas to be created before their environment variables are updated.

Reply:
I meant that we need to club it all together instead of having 2 if conditions for props.enableKendra in the same file

Refer [this](https://github.com/awslabs/document-understanding-solution/pull/46#discussion_r438482751) for more details